typescript/cli/src/deploy/warp.ts

```diff
@@ -132,6 +156,71 @@ async function executeDeploy(params: DeployParams) {
   logBlue('Deployment is complete!');
 }
 
+async function fetchOrDeployIsmFactoryAddresses(
+  config: WarpRouteDeployConfig,
+  registry: IRegistry,
+  ismFactoryDeployer: HyperlaneProxyFactoryDeployer,
+): Promise<HyperlaneAddressesMap<any>> {
+  return promiseObjAll(
+    objMap(config, async (chain, _config) => {
+      const chainAddresses = await registry.getChainAddresses(chain);
+      if (chainAddresses) {
+        logGray('Registry factory addresses loaded');
+        return chainAddresses; // Can includes other addresses
+      } else {
+        logGray('Registry factory addresses not found, deploying');
+        return serializeContracts(
+          await ismFactoryDeployer.deployContracts(chain),
+        );
+      }
+    }),
+  );
+}
+
+/**
+ * Deploys the Warp ISM for each chain in the provided `deployedFactoriesAddresses`.
+ * @returns IsmConfig with each Ism as an address string
+ */
+async function deployWarpIsm(
+  config: WarpRouteDeployConfig,
+  multiProvider: MultiProvider,
+  deployedFactoriesAddresses: HyperlaneAddressesMap<any>,
+  ismFactoryDeployer: HyperlaneDeployer<any, any>,
```

Comment:
TODO: Fix `any` types

Reply:
not landed yet but something I was working on for hook deploys https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3861/commits/a68b5a0746f5296c3017b922f83ca59f80003be3#diff-583dd89cdc1b35595ab28c69c5f1a59a156a9c546cd5ef4fd0e3a3a52c7393aaL86

Reply:
> TODO: Fix `any` types

\+